sycl/test/ordered_queue/oq_kernels.cpp

```diff
@@ -0,0 +1,67 @@
+// RUN: %clangxx -fsycl -fsycl-unnamed-lambda %s -o %t.out
+// RUN: env SYCL_DEVICE_TYPE=HOST %t.out
+// RUN: %ACC_RUN_PLACEHOLDER %t.out
+// RUN: %CPU_RUN_PLACEHOLDER %t.out
+// RUN: %GPU_RUN_PLACEHOLDER %t.out
+
+//==------ oq_kernels.cpp - SYCL ordered queue kernel shortcut test --------==//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+
+#include <CL/sycl.hpp>
+#include <iostream>
+using namespace cl::sycl;
+
+int main() {
+  ordered_queue q;
+  auto dev = q.get_device();
+  auto ctx = q.get_context();
+  const int N = 8;
+
+  auto A = (int *)malloc_shared(N * sizeof(int), dev, ctx);
```

Comment:
While doing a presentation yesterday I was thinking also to a more allocator-related syntax, such as:
```
   auto A = malloc_shared<int>(N * sizeof(int), dev, ctx);
```

Reply:
This is not related to this patch, so I suggest filing separate issue/feature request for James.